torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -452,22 +452,22 @@ std::vector<int64_t> ComputeShapeFromReshape(
   }
   std::vector<int64_t> final_shape;
   // shape_ratio is used to calculate the real value of -1.
-  // it is updated on each dimension to avoid overflow.
   double shape_ratio = 1.0;
   for (auto i = 0; i < input_shape_size; i++) {
+    shape_ratio *= static_cast<double>(input_shape[i]);
+  }
+  for (auto i = 0; i < reshape_size; i++) {
     if (i < minus_one_pos) {
       if (reshape[i] != 0) {
-        shape_ratio *= static_cast<double>(input_shape[i]) / reshape[i];
+        shape_ratio /= static_cast<double>(reshape[i]);
+      } else {
+        shape_ratio /= static_cast<double>(input_shape[i]);
       }
-    } else if (minus_one_pos > -1) {
-      if (i < input_shape_size - reshape_size + minus_one_pos + 1) {
-        shape_ratio *= static_cast<double>(input_shape[i]);
+    } else if (i > minus_one_pos) {
+      if (reshape[i] != 0) {
```

Comment:
Just curious as to what are cases in which reshape[dim] = 0, Are these cases particular to onnx::reshape nodes from the symbolics?

Reply:
Seems that pytorch exporter does not have reshape[dim]==0, this is for onnx.

Reply:
As expected. Thanks for confirming.